### ISSUE TITLE
snabbvmx is called without the --mirror knob

### ISSUE THREAD

**Issue by chrgraf** [Juniper/vmx-docker-lwaftr#1]
Dear all,

snabb lwaftr provides a monitor app. Due to missing items below this "monitor" app is non-functional.

To use the monitor app, snabbvmx must be called with a "--mirror tap\<x\>" knob. This --mirror knob is missing.
As well the tap-interfaces (one per each snabb instance) inside the container needs to be created.

Please have the tap-interfaces and the "--mirror" knob added to the launcher scripts.

All details can be found here: https://github.com/Igalia/snabb/issues/579

Input from Marcel:
- The original version of snabbvmx always connected to a mirror port with the same name as the id given to snabbvmx. The final version added a new parameter '--mirror' that needs to be added to the launch script in the vmx-docker-lwaftr container.
- he indicated that this might be the place to add the "--mirror" knob when launching the "snabb snabbvmx":https://github.com/Juniper/vmx-docker-lwaftr/blob/master/launch_snabb.sh#L48

I do have a setup to reproduce and demonstrate the issue.

kindly have the issue fixed. 

thanks

christian

**Comment by mwiget** [comment]
tap interfaces are already created in the container:
See https://github.com/Juniper/vmx-docker-lwaftr/blob/master/launch.sh#L516-L518
They have names like 'xe0', 'xe1', etc.

Only the knob '--mirror' is missing.

**Comment by chrgraf** [comment]
Dear all,

I have been rebuilding the docker image and applied Marcel's change to launch_snabb.sh.
Not sure if both lines below or only one line below finally requires the --mirror...

```
lab@ubuntu1:~/vmx-docker-lwaftr$ grep mirror launch_snabb.sh
sudo usr/local/bin/snabb snabbvmx lwaftr --conf test-snabbvmx-lwaftr-${INT}.cfg --id $INT --pci $PCI --mac $MAC --mirror xe0
  CMD="$NUMACTL $SNABB snabbvmx lwaftr --conf snabbvmx-lwaftr-${INT}.cfg --id $INT --pci $PCI --mac $MAC --sock %s.socket --mirror xe0"
```

As Marcel pointed out the tap-interfaces  (xe0,xe1,...xe<n>) are already being created. No change required on those.

The above changes by adding the "--mirror" result in snabb getting properly launched:
```
root@34fd0efe002c:/# ps -ef | grep mirr
root       469   222  0 08:27 ?        00:00:00 /usr/local/bin/snabb snabbvmx lwaftr --conf snabbvmx-lwaftr-xe0.cfg --id xe0 --pci 0000:81:00.0 --mac 02:cf:69:15:81:00 --sock %s.socket --mirror xe0
root       471   469 99 08:27 ?        00:12:27 /usr/local/bin/snabb snabbvmx lwaftr --conf snabbvmx-lwaftr-xe0.cfg --id xe0 --pci 0000:81:00.0 --mac 02:cf:69:15:81:00 --sock %s.socket --mirror xe0
```

the monitor-function is fully functional

```
root@34fd0efe002c:/# snabb lwaftr monitor all
Monitor all
Mirror address set to 'all' in PID '471'
root@34fd0efe002c:/# tcpdump -n -i xe0
tcpdump: WARNING: xe0: no IPv4 address assigned
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on xe0, link-type EN10MB (Ethernet), capture size 65535 bytes
08:28:52.537716 IP6 :: > ff02::16: HBH ICMP6, multicast listener report v2, 1 group record(s), length 28
```

**Note:**
 Above fix is not complete. This just works for a single snabb instance. If multiple snabb instances are in use, then each instance requires to be assigned to its dedicated xe<n> tap interface.

thanks

christian

**Comment by mwiget** [comment]
fixed and tagged in v1.1.20
